src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -107,23 +131,29 @@ internal static async Task<int> Run(PackageUpdateArgs args, ILoggerWithColor log
         // 4. Update MSBuild files
         var projectName = Path.GetFileNameWithoutExtension(dgSpec.Projects[0].FilePath);
         logger.LogInformation($"  {projectName}:");
-        logger.LogInformation($"    " + Format.PackageUpdate_UpdatedMessage(packageToUpdate.Id, packageToUpdate.CurrentVersion.ToShortString(), packageToUpdate.NewVersion.ToShortString()));
-        logger.LogInformation("");
 
         var updatedPackageSpec = updatedDgSpec.Projects[0];
-        packageUpdateIO.UpdatePackageReference(updatedPackageSpec, restorePreviewResult, packageTfms!, packageToUpdate, logger);
+        int updatedCount = 0;
+
+        foreach (var packageResult in packagesToUpdateResult)
+        {
+            logger.LogInformation($"    " + Format.PackageUpdate_UpdatedMessage(packageResult.Package.Id, packageResult.Package.CurrentVersion.ToShortString(), packageResult.Package.NewVersion.ToShortString()));
+            packageUpdateIO.UpdatePackageReference(updatedPackageSpec, restorePreviewResult, packageResult.Frameworks, packageResult.Package, logger);
```

Comment:
UpdatePackageReference itself should probably actually be passing aliases. 

I think CPS package updates already use aliasing, so I think you need the same thing here. 

It's possible dotnet add package has a bug, but I'll let you verify that :D

Reply:
I made IPackageUpdateIO use TFM aliases, but it calls into add package APIs to actually modify the project file. I'm going to consider that out of scope for package update (I need to focus package update features to make .NET 10 deadlines).

Since PackageSpec only has a `GetTargetFramework(NuGetFramework)` API, and no `GetTargetFramework(string)` API, it'd be good for us to try to remember to change PackageUpdateIO.cs when such an API is added.